lib/compat/wordpress-5.9/edit-site-page.php

```diff
@@ -114,12 +114,13 @@ static function( $classes ) {
 	}
 
 	$custom_settings = array(
-		'siteUrl'                  => site_url(),
-		'postsPerPage'             => get_option( 'posts_per_page' ),
-		'styles'                   => gutenberg_get_editor_styles(),
-		'defaultTemplateTypes'     => $indexed_template_types,
-		'defaultTemplatePartAreas' => get_allowed_block_template_part_areas(),
-		'__unstableHomeTemplate'   => gutenberg_resolve_home_template(),
+		'siteUrl'                         => site_url(),
+		'postsPerPage'                    => get_option( 'posts_per_page' ),
+		'styles'                          => gutenberg_get_editor_styles(),
+		'defaultTemplateTypes'            => $indexed_template_types,
+		'defaultTemplatePartAreas'        => get_allowed_block_template_part_areas(),
+		'__unstableHomeTemplate'          => gutenberg_resolve_home_template(),
+		'__unstableThemeJSONLatestSchema' => WP_Theme_JSON_Gutenberg::LATEST_SCHEMA,
```

Comment:
Should this change live in a 6.1 compat file?

Reply:
I'd defer to @adamziel on that one. I am behind with how all this is supposed to work. Id love to see some docs on it if there is somewhere I can reference.

Reply:
Docs would be awesome, I am getting confused here all the time! The basic idea is:

* Newer Gutenberg versions can run on older WordPress versions
* If that older WordPress is missing any features, Gutenberg "polyfills" them via code living in `lib/compat`.
* If this code is needed for Gutenberg 13.2 to run and will only be merged in WordPress 6.1, then it should live in lib/compat

In this case, you want:
* WordPress 5.9 to use this change
* WordPress 6.0 to use this change
* WordPress 6.1 to use this change

It seems like this file is loaded in all these releases:

https://github.com/WordPress/gutenberg/blob/d5b06563e8acb3f4597731fe3faa4b860fb62a26/lib/load.php#L99

It also seems like this code has a counterpart in wordpress-develop:

https://github.com/WordPress/wordpress-develop/blob/04e9728701b3dde0260c7cdebfd726edbe62ac97/src/wp-admin/site-editor.php#L63-L70

So I'm confused, does this mean that Gutenberg plugin always replaces that entire page with a custom one? If yes, then updating `lib/compat/wordpress-5.9` + a backport to wordpress-develop would likely be sufficient. You could even move that entire function to `lib/compat/wordpress-6.1` to make it clear to 6.1 release leads that there were relevant changes here

Reply:
@scruffian I've played a bit with a version matrix and I think this is what we're after here:

❌ – fix not included
✅ – fix included

| WP\Plugin   | No plugin | 13.2        | 13.3 |
|-------------|-----------|-------------|------|
| 5.8 (10.7)  | ❌         | ❌           | ✅    |
| 5.9 (11.9)  | ❌         | ❌           | ✅    |
| 6.0 (13.1)  | ❌         | ❌           | ✅    |
| 6.1 (13.3+) | ✅         | Unsupported 🤞  | ✅    |

Which means two things need to happen:

1) Your patch gets included in Gutenberg 13.3
2) It then gets backported to WordPress 6.1

This makes me think you could move this entire function to `lib/compat/wordpress-6.1` and apply your patch there.